### PR TITLE
add search product

### PR DESCRIPTION
**Pull request by DariaLavrushko**

### CODE REVIEW

File: src/main/java/de/allcom/services/ProductService.java [hunk 1]
```diff
@@ -100,59 +100,24 @@ public ProductResponseValues saveProduct(SaveProductRequestDto saveProductReques
...
 
     public ProductResponseValues findById(Long id) {
         Product product = productRepository.findById(id)
-                .orElseThrow(() -> new RestException(HttpStatus.NOT_FOUND, "Product whith id: "
+                .orElseThrow(() -> new RestException(HttpStatus.NOT_FOUND, "Product with id: "
                         + id + " did not found"));
+        return mapFrom(product);
+    }
+
+    private ProductResponseValues mapFrom(Product product) {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java]: - move to ProductResponseValues
- just `from`

Comment (Igor-Klein-1988) [reply src/main/java/de/allcom/services/ProductService.java]: see `UserDto`

File: src/main/java/de/allcom/controllers/api/ProductApi.java [hunk 2]
```diff
@@ -39,27 +40,27 @@ public interface ProductApi {
...
-    @Operation(summary = "Product add", description = "Default role is Admin")
+
+    @Operation(summary = "Product info", description = "Default role is Client")
     @ApiResponses(value = {
             @ApiResponse(responseCode = "200", description = "Product found", content =
             @Content(mediaType = "application/json", schema = @Schema(implementation = ProductDto.class))),
-            @ApiResponse(responseCode = "404", description = "Product did not found", content =
-            @Content(mediaType = "application/json", schema = @Schema(implementation = ProductDto.class)))
+            @ApiResponse(responseCode = "400", description = "Bad request", content =
+            @Content(mediaType = "application/json", schema = @Schema(implementation = CategoryDto.class)))
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/controllers/api/ProductApi.java]: StandartResponseDto
check 48 line too

File: src/main/java/de/allcom/services/ProductService.java [hunk 3]
```diff
@@ -100,80 +98,36 @@ public ProductResponseValues saveProduct(SaveProductRequestDto saveProductReques
...
+    public Page<ProductResponseValues> searchByCategoryOrName(Long categoryId, String searchQuery,
+                                                              PageRequest pageRequest) {
+        Page<Product> products;
 
+        if (categoryId != null) {
+            Category category = categoryRepository.findById(categoryId)
+                    .orElseThrow(() -> new RestException(HttpStatus.BAD_REQUEST,
+                            "Category with id: " + categoryId + " not found"));
 
+            products = searchQuery != null && !searchQuery.isEmpty()
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java]: add variable `boolean isSearchQueryPresent = searchQuery != null && !searchQuery.isEmpty();`

Comment (Igor-Klein-1988) [reply src/main/java/de/allcom/services/ProductService.java]: and use here

File: src/main/java/de/allcom/services/ProductService.java [hunk 4]
```diff
@@ -100,80 +98,36 @@ public ProductResponseValues saveProduct(SaveProductRequestDto saveProductReques
...
+                    ? productRepository.findAllByCategoryAndNameStartsWithIgnoreCase(category, searchQuery, pageRequest)
+                    : productRepository.findAllByCategory(category, pageRequest);
+        } else {
+            products = searchQuery != null && !searchQuery.isEmpty()
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java]: and use here

**Review by Igor-Klein-1988 — APPROVED** [review]

**Review by maiorovvv — APPROVED** [review]

**Pull request merged**